app/__init__.py

```diff
@@ -53,4 +74,14 @@ def remove_trailing_slash():
             else:
                 return redirect(request.path[:-1], code=301)
 
+    @application.before_request
+    def refresh_session():
+        session.permanent = True
+        session.modified = True
```

Comment:
It's a straight copy from the supplier app but I believe that having `session.modified = True` effectively resets the session timer with each page view, so it works as "time since last page view" rather than "time since login".

But I could be wrong about that.